### PR TITLE
Document support for insecure LDAP connection

### PR DESCRIPTION
**Pull request by lukasz-walkiewicz**

### CODE REVIEW

File: presto-docs/src/main/sphinx/security/ldap.rst [hunk 1]
```diff
@@ -2,12 +2,11 @@
...
 ===================
 
-Presto can be configured to enable frontend LDAP authentication over
-HTTPS for clients, such as the :ref:`cli_ldap`, or the JDBC and ODBC
-drivers. At present, only simple LDAP authentication mechanism involving
-username and password is supported. The Presto client sends a username
-and password to the coordinator, and the coordinator validates these
-credentials using an external LDAP service.
+Presto can be configured to enable frontend LDAP authentication for clients,
+such as the :ref:`cli_ldap`, or the JDBC and ODBC drivers. At present, only
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/security/ldap.rst:6]: Remove ODBC driver since there is no open source ODBC driver. Make the JDBC driver mention a link.

Comment (mosabua) [inline presto-docs/src/main/sphinx/security/ldap.rst:6]: Also remove "At present,"

File: presto-docs/src/main/sphinx/security/ldap.rst [hunk 2]
```diff
@@ -96,7 +89,10 @@ Password authentication needs to be configured to use LDAP. Create an
 Property                                                Description
 ======================================================= ======================================================
 ``ldap.url``                                            The url to the LDAP server. The url scheme must be
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/security/ldap.rst:91]: The URL to the LDAP server. The URL scheme must be ...

File: presto-docs/src/main/sphinx/security/ldap.rst [hunk 3]
```diff
@@ -96,7 +89,10 @@ Password authentication needs to be configured to use LDAP. Create an
...
-                                                        ``ldaps://`` since Presto allows only Secure LDAP.
+                                                        ``ldap://`` or ``ldaps://``. Connecting to the LDAP
+                                                        server without SSL enabled requires
+                                                        ``ldap.allow-insecure=true``.
+``ldap.allow-insecure``                                 Allow insecure connection to the LDAP server.
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/security/ldap.rst:95]: This property should be renamed... insecure does NOT mean the same as "not secured" .. https://www.merriam-webster.com/dictionary/insecure. At least the description should change.

Comment (mosabua) [reply presto-docs/src/main/sphinx/security/ldap.rst:95]: Maybe something like "Allow using an LDAP connection that is not secured with TLS."

File: presto-docs/src/main/sphinx/security/ldap.rst [hunk 4]
```diff
@@ -2,12 +2,11 @@
 LDAP Authentication
 ===================
 
-Presto can be configured to enable frontend LDAP authentication over
```
Comment (electrum) [inline presto-docs/src/main/sphinx/security/ldap.rst:5]: Revert this change. The HTTPS is referring to the requirement on the Presto coordinator, not the backend connection to LDAP.

File: presto-docs/src/main/sphinx/security/ldap.rst (hunk 1 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/security/ldap.rst:6]: Let’s leave this alone for now. Changing the text here for ODBC is out of scope for unsecured LDAP.

File: presto-docs/src/main/sphinx/security/ldap.rst (hunk 3 above)
Comment (lukasz-walkiewicz) [reply presto-docs/src/main/sphinx/security/ldap.rst:95]: I'm not an expert in this area but it's pretty popular naming convention, eg. chrome has `allow-insecure-localhost`.

Comment (mosabua) [reply presto-docs/src/main/sphinx/security/ldap.rst]: popular doesnt mean its correct. I would try to avoid since its plain wrong.

Comment (lukasz-walkiewicz) [reply presto-docs/src/main/sphinx/security/ldap.rst]: @electrum @kokosing wdyt?

**Review by kokosing — APPROVED** [review]

File: presto-docs/src/main/sphinx/security/ldap.rst [hunk 5]
```diff
@@ -21,12 +21,6 @@ Presto Server Configuration
 Environment Configuration
```
Comment (dain) [inline presto-docs/src/main/sphinx/security/ldap.rst:21]: This section is now empty and should be removed

**Comment by dain** [comment]
@lukasz-walkiewicz can you fix the bad property name for forwarding while you are at this? The correct name is `http-server.process-forwarded`

**Review by kokosing — APPROVED** [review]
